src/main/java/org/jvnet/hudson/update_center/Plugin.java

```diff
@@ -166,12 +166,15 @@ public String getPluginUrl() throws IOException {
         if (url == null) {
             url = readSingleValueFromXmlFile(latest.resolvePOM(), "/project/url");
         }
-
+        // last fallback: GitHub URL; also prevent plugins.j.io referencing itself
+        if (url == null || url.startsWith("https://plugins.jenkins.io")) {
+            url = requireTopLevelUrl(getScmUrl());
+        }
         String originalUrl = url;
 
         if (url != null) {
             url = url.replace("wiki.hudson-ci.org/display/HUDSON/", "wiki.jenkins-ci.org/display/JENKINS/");
-            url = url.replace("http://wiki.jenkins-ci.org", "https://wiki.jenkins-ci.org");
+            url = url.replace("http://wiki.jenkins-ci.org", "https://wiki.jenkins.io");
```

Comment:
This unrelated improvement made it unnecessarily cumbersome to review the impact of this change on the update center output. In the future, please separate out unnecessary changes.